### PR TITLE
Change all instances of JobID to DriverID.

### PR DESCRIPTION
**Pull request by justinwyang**

### CODE REVIEW

**Comment by jovany-wang** [comment]
I think JobID is different from DriverID. In the future, driver may exit quickly and the job are still running in the cluster.

**Comment by robertnishihara** [comment]
@jovany-wang, so far we have been using them interchangeably.

Even in the scenario that you mention, where a driver starts and then exits, but certain tasks/actors created by the driver continue running, the concepts still seem the same to me. E.g., the "job" consists of all tasks/actors created by a given "driver" script. So there is a one-to-one mapping between `DriverID`s and `JobID`s. Given that these concepts seem to represent the same thing, I think we should make them a single concept.

File: src/ray/gcs/client_test.cc [hunk 1]
```diff
@@ -121,15 +121,15 @@ void TestTableLookup(const JobID &job_id, std::shared_ptr<gcs::AsyncGcsClient> c
 #define TEST_MACRO(FIXTURE, TEST) \
   TEST_F(FIXTURE, TEST) {         \
     test = this;                  \
-    TEST(job_id_, client_);       \
+    TEST(driver_id_, client_);       \
```
Comment (robertnishihara) [inline src/ray/gcs/client_test.cc]: @justinwyang please fix the indentation here

File: src/ray/raylet/raylet_client.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#ifndef RAYLET_CLIENT_H
+DriverID#ifndef RAYLET_CLIENT_H
```
Comment (robertnishihara) [inline src/ray/raylet/raylet_client.h]: This looks like a mistake.

**Comment by robertnishihara** [comment]
@justinwyang please fix linting and other test failures.

File: python/ray/tests/test_multi_node.py [hunk 3]
```diff
@@ -317,7 +317,7 @@ def test_calling_start_ray_head():
             "ray", "start", "--head", "--redis-port", "6379",
             "--redis-shard-ports", "6380,6381,6382", "--object-manager-port",
             "12345", "--num-cpus", "2", "--num-gpus", "0",
-            "--redis-max-clients", "100", "--resources", "{\"Custom\": 1}"
+            "--redis-max-clients", "100", "--resources", "{\'Custom\': 1}"
```
Comment (jovany-wang) [inline python/ray/tests/test_multi_node.py]: I don't think this change is correct. We shouldn't make sure the json parser treats `\'`  as  ` \" `.

I think the best approach to fix the CI failure is disable the check right now, then enable the quote check which can allow this statement `"\"abc\""` and reject this kind statement `'abc'`.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**